token-account-based/chaincode-go/chaincode/token_contract.go

```diff
@@ -58,6 +61,28 @@ func (s *SmartContract) Mint(ctx contractapi.TransactionContextInterface, amount
 
 	log.Printf("minter account %s balance updated from %d to %d", minter, currentBalance, updatedBalance)
 
+	// Update the totalSupply
+	totalSupplyBytes, err := ctx.GetStub().GetState(totalSupplyKey)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve total token supply: %v", err)
+	}
+
+	var totalSupply int
+
+	// If no tokens have been minted, initialize the totalSupply
+	if totalSupplyBytes == nil {
+		totalSupply = 0
+	} else {
+		totalSupply, err = strconv.Atoi(string(totalSupplyBytes))
+		if err != nil {
+			return fmt.Errorf("failed to convert ascii string %s to integer: %v", totalSupplyBytes, err)
```

Comment:
While it could be open for debate, throughout this contract I did not handle the Atoi() errors and instead put messages like:

// Error handling not needed since Itoa() was used when setting the account balance, guaranteeing it was an integer.

The intent was to keep the code as concise as possible without extra error handling getting in the way. I'd suggest to stick with the precedent that is already in this file.